lib/ddtrace/tracer.rb

```diff
@@ -100,13 +101,19 @@ def configure(options = {})
     #
     #   tracer.set_service_info('web-application', 'rails', 'web')
     def set_service_info(service, app, app_type)
-      @services[service] = {
+      service_key = if services_prefix && service.start_with?(services_prefix)
```

Comment:
I think this is "not enough" as `set_service_info` is not called that often. In fact, you could trace a lot of code and never call it as it's only used to update the metadata of the service (eg: say "hey, this service is a database"). So I think to complete this patch one would need to hook somewhere in the `service=` method of spans (which is implemented with an `attr_accessor` right now). But as you mention, this is related to #142 so let's save this for now I think.